packages/@aws-cdk/aws-logs-destinations/test/lambda.test.ts

```diff
@@ -79,14 +79,10 @@ test('lambda permissions are not added when addPermissions is false', () => {
   });
 
   // THEN: subscription target is Lambda
-  expect(stack).toHaveResource('AWS::Logs::SubscriptionFilter', {
+  Template.fromStack(stack).hasResourceProperties('AWS::Logs::SubscriptionFilter', {
     DestinationArn: { 'Fn::GetAtt': ['MyLambdaCCE802FB', 'Arn'] },
   });
 
   // THEN: Lambda does not have permissions to be invoked by CWL
-  expect(stack).not.toHaveResource('AWS::Lambda::Permission', {
-    Action: 'lambda:InvokeFunction',
-    FunctionName: { 'Fn::GetAtt': ['MyLambdaCCE802FB', 'Arn'] },
-    Principal: 'logs.amazonaws.com',
-  });
+  Template.fromStack(stack).resourceCountIs('AWS::Lambda::Permission', 0);
```

Comment:
Same comment as before. These assertions don't meant the same thing.

Reply:
The current behaviour here - 

```ts
Template.fromStack(stack).hasResourceProperties('AWS::Lambda::Permission', Match.not({
  ...
});
```

will fail with `No resource of type AWS::Lambda::Permission` so this is the best alternative.

Tracking this here - https://github.com/aws/aws-cdk/issues/15868